crates/polars-time/src/chunkedarray/date.rs

```diff
@@ -84,19 +84,32 @@ pub trait DateMethods: AsDate {
         day: &Int8Chunked,
         name: PlSmallStr,
     ) -> PolarsResult<DateChunked> {
+        let mut error_values: Option<(i32, i8, i8)> = None;
         let mut ca: Int32Chunked = year
             .into_iter()
             .zip(month)
             .zip(day)
             .map(|((y, m), d)| {
                 if let (Some(y), Some(m), Some(d)) = (y, m, d) {
-                    NaiveDate::from_ymd_opt(y, m as u32, d as u32)
-                        .map(|t| t.num_days_from_ce() - EPOCH_DAYS_FROM_CE)
+                    NaiveDate::from_ymd_opt(y, m as u32, d as u32).map_or_else(
+                        // If None is returned, then we have an invalid date.
+                        // We save the faulty values and move on.
+                        || {
+                            error_values = Some((y, m, d));
+                            None
+                        },
+                        // We have a valid date.
+                        |ns| Some(ns.num_days_from_ce() - EPOCH_DAYS_FROM_CE),
+                    )
                 } else {
                     None
                 }
             })
             .collect_trusted();
```

Comment:
You can return a result and then call `try_collect_ca_with_dtype`, that's a bit cleaner.